diff --git a/Librarys/AIMacro/Library.AIMacroCodeGen4.ailang b/Librarys/AIMacro/Library.AIMacroCodeGen4.ailang
index 2babcd2..6045c0a 100644
--- a/Librarys/AIMacro/Library.AIMacroCodeGen4.ailang
+++ b/Librarys/AIMacro/Library.AIMacroCodeGen4.ailang
@@ -1537,6 +1537,12 @@ Function.Gen_MapBuiltin {
         cmp = StringCompare(name, "type")
         IfCondition EqualTo(cmp, 0) ThenBlock: { ReturnValue("Types.TypeOf") }
 
+        cmp = StringCompare(name, "frozenset")
+        IfCondition EqualTo(cmp, 0) ThenBlock: { ReturnValue("AIMacro.FrozenSet") }
+
+        cmp = StringCompare(name, "set")
+        IfCondition EqualTo(cmp, 0) ThenBlock: { ReturnValue("AIMacro.MakeSet") }
+
         ReturnValue(name)
     }
 }
